src/uwtools/drivers/mpas_init.py

```diff
@@ -0,0 +1,178 @@
+"""
+A driver for the mpas-init component.
+"""
+
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Any, Dict
+
+from iotaa import asset, dryrun, task, tasks
+
+from uwtools.api.template import render
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file, filecopy, symlink
+
+
+class MPASInit(Driver):
+    """
+    A driver for mpas-init.
+    """
+
+    def __init__(self, config: Path, cycle: datetime, dry_run: bool = False, batch: bool = False):
+        """
+        The driver.
+
+        :param config_file: Path to config file (read stdin if missing or None).
+        :param cycle: The cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config=config, dry_run=dry_run, batch=batch, cycle=cycle)
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @tasks
+    def boundary_files(self):
+        """
+        Boundary-condition files.
+        """
+        yield self._taskname("boundary files")
+        lbcs = self._driver_config["boundary_conditions"]
+        endhour = lbcs["length"]
+        interval = lbcs["interval_hours"]
+        symlinks = {}
+        boundary_filepath = lbcs["path"]
+        for boundary_hour in range(0, endhour + 1, interval):
+            file_date = self._cycle + timedelta(hours=boundary_hour)
+            target = Path(boundary_filepath) / f"FILE:{file_date.strftime('%Y-%m-%d_%H')}"
+            linkname = self._rundir / f"FILE:{file_date.strftime('%Y-%m-%d_%H')}"
+            symlinks[target] = linkname
+        yield [symlink(target=t, linkname=l) for t, l in symlinks.items()]
```

Comment:
I'm partial to the `fn` use in the #448 , but we should update the fv3 driver with it as well. I'd almost want to add `offset=0` for consistency too.

Reply:
I agree that the `fn` might be one way to DRY the mpas-init code, but otherwise I think these two are doing what they should for their given drivers. 

For mpas_init, it prepares boundary conditions, both lateral and boundary. In mpas, all the files that are needed are named `lbc.*.nc`, so it seemed more consistent to name it specifically `lateral_boundary_conditions`.

For mpas_init, it needs to know how many boundary conditions to prepare, so we included `length` as a setting for the boundary conditions. For mpas, the length is of the forecast itself, so we chose to stay consistent with the `fv3` driver and have `length` live under the driver config.

Reply:
I'm also bothered by the apparent duplication here, though I believe you (but haven't read closely enough to fully understand) that there are important differences. Still, I wonder if some generalization could be done later (not in this PR) to DRY this out. If I'm not just wrong, can we at least create a ticket to consider refactoring this later?